aten/src/ATen/native/ReflectionPad.cpp

```diff
@@ -686,5 +976,120 @@ Tensor reflection_pad2d_backward_cpu(
   return grad_input;
 }
 
+TORCH_IMPL_FUNC(reflection_pad3d_out_cpu)
+(const Tensor& input_, IntArrayRef padding, const Tensor& output) {
+  int64_t pad_left = padding[0];
+  int64_t pad_top = padding[2];
+  int64_t pad_front = padding[4];
+  int64_t dim_w = 3;
+  int64_t dim_h = 2;
+  int64_t dim_d = 1;
+  int64_t dim_plane = 0;
+  bool batch_mode = (input_.dim() == 5);
+
+  if (batch_mode) {
+    dim_w++;
+    dim_h++;
+    dim_d++;
+    dim_plane++;
+  }
+
+  int64_t nplane = input_.size(dim_plane);
+  int64_t input_w = input_.size(dim_w);
+  int64_t input_h = input_.size(dim_h);
+  int64_t input_d = input_.size(dim_d);
+  int64_t output_w = output.size(dim_w);
+  int64_t output_h = output.size(dim_h);
+  int64_t output_d = output.size(dim_d);
+
+  auto input = input_.contiguous();
+
+  if (batch_mode) {
+    AT_DISPATCH_FLOATING_AND_COMPLEX_TYPES(input.scalar_type(), "replication_pad3d_cpu", [&] {
+      auto input_data = input.data_ptr<scalar_t>();
+      auto output_data = output.data_ptr<scalar_t>();
+      auto nbatch = input.size(0);
+      reflection_pad3d_out_loop(
+          input_data, output_data,
+          nbatch, nplane,
+          input_w, input_h, input_d,
+          output_w, output_h, output_d,
+          pad_left, pad_top, pad_front);
+      });
+  } else {
+    AT_DISPATCH_FLOATING_AND_COMPLEX_TYPES(input.scalar_type(), "replication_pad3d_cpu", [&] {
+      auto input_data = input.data_ptr<scalar_t>();
+      auto output_data = output.data_ptr<scalar_t>();
+      reflection_pad3d_out_frame(
+          input_data, output_data,
+          nplane,
+          input_w, input_h, input_d,
+          output_w, output_h, output_d,
+          pad_left, pad_top, pad_front);
+    });
+  }
+}
+
+TORCH_IMPL_FUNC(reflection_pad3d_backward_out_cpu)(const Tensor& grad_output,
+    const Tensor& input,
+    IntArrayRef padding,
+    const Tensor& grad_input) {
+  int64_t pad_left = padding[0];
+  int64_t pad_top = padding[2];
+  int64_t pad_front = padding[4];
+  int64_t dim_w = 3;
+  int64_t dim_h = 2;
+  int64_t dim_d = 1;
+  int64_t dim_plane = 0;
+  bool batch_mode = (input.dim() == 5);
```

Comment:
Please add `TORCH_CHECK(input.dim() >= 4)` and `TORCH_CHECK(grad_output.dim() == input.dim());`
```suggestion
  bool batch_mode = (input.dim() == 5);
  TORCH_CHECK(input.dim() > 3);
  TORCH_CHECK(grad_output.dim() == input.dim());
  TORCH_CHECK(grad_input.dim() == grad_output.dim());
```